lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -654,6 +654,24 @@ protected function getPreAlterTableIndexForeignKeySQL(TableDiff $diff)
             }
         }
 
+        // handle columns that are removed from changed indexes
+        foreach ($diff->changedIndexes as $chgKey => $chgIndex) {
+            // Dropping primary keys requires to unset autoincrement attribute on the particular column first
+            if ($chgIndex->isPrimary() && $diff->fromTable instanceof Table) {
+
+                foreach ($diff->fromTable->getIndex($chgIndex->getName())->getColumns() as $columnName) {
```

Comment:
Can you use `$diff->fromTable->getPrimaryKey()` here? It is more explicit and the correct API for retrieving it and does not rely on the name of a primary key.